### PR TITLE
OTT-1633: OW Bug Fix: Do not send null device.ext.atts

### PR DESCRIPTION
**Pull request by ashishshinde-pubm**
We need to contribute this changes to Prebid open source.

### CODE REVIEW

File: modules/pubmatic/openwrap/models/device.go [hunk 1]
```diff
@@ -56,8 +59,90 @@ const (
...
 type ExtDevice struct {
-	openrtb_ext.ExtDevice
-	SessionID string `json:"session_id,omitempty"`
-	IDFV      string `json:"idfv,omitempty"`
+	data map[string]any
+}
+
+func (e *ExtDevice) Init() {
+	e.data = make(map[string]any)
+}
```
Comment (pm-nilesh-chate) [inline modules/pubmatic/openwrap/models/device.go:76]: func NewDeviceExt() *ExtDevice {
      return ExtDevice{data: make(map[string]any)}
}

File: modules/pubmatic/openwrap/models/device.go [hunk 2]
```diff
@@ -56,8 +59,90 @@ const (
...
+	}
+	val, found := e.data[key]
+	if !found {
+		return 0, found
+	}
+	value, found = val.(float64)
+	return value, found
+}
+
+func (e *ExtDevice) GetAtts() (value float64, found bool) {
```
Comment (pm-nilesh-chate) [inline modules/pubmatic/openwrap/models/device.go:123]: make all return type *float if callers are using pointers

**Review by pm-nilesh-chate — DISMISSED** [review]

File: modules/pubmatic/openwrap/device.go [hunk 3]
```diff
@@ -36,29 +35,45 @@ func updateDeviceIFADetails(dvc *models.DeviceCtx) {
...
-		if dvc.DeviceIFA != "" {
-			if _, ok := models.DeviceIFATypeID[strings.ToLower(deviceExt.IFAType)]; !ok {
-				deviceExt.IFAType = ""
-			}
-		} else if deviceExt.SessionID != "" {
-			dvc.DeviceIFA = deviceExt.SessionID
-			deviceExt.IFAType = models.DeviceIFATypeSESSIONID
-		} else {
-			deviceExt.IFAType = ""
+	extIFATypeStr, _ := deviceExt.GetIFAType()
```
Comment (pm-viral-vala) [inline modules/pubmatic/openwrap/device.go:38]: Please check all test cases are working without modifying actual and expected values.

File: modules/pubmatic/openwrap/models/device.go (hunk 1 above)
Comment (ashishshinde-pubm) [reply modules/pubmatic/openwrap/models/device.go]: done

**Review by pm-viral-vala — APPROVED** [review]

**Pull request merged**